src/Symfony/Bundle/DependencyInjection/Configuration.php

```diff
@@ -163,6 +163,12 @@ public function getConfigTreeBuilder(): TreeBuilder
         $this->addFormatSection($rootNode, 'patch_formats', [
             'json' => ['mime_types' => ['application/merge-patch+json']],
         ]);
+        $this->addFormatSection($rootNode, 'docs_formats', [
```

Comment:
We also need to do something about the `formats` section. We should deprecate not setting explicitly `formats` in 3.x and remove `json` from the default values in 4.0.

If `formats` is not set, then we should enable `json` for docs endpoint for BC.

I would remove YAML from the defaults. IMHO it should only be enabled explicitly by the user because it's slower and it increases the attack surface.

The recipe must also be updated to explicitly set these two keys.